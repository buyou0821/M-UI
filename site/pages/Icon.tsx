import React from 'react';
import { Icon } from 'components/index';

export default () => {
  const IconFont = Icon.createFromIconfont({
    scriptUrl: '//at.alicdn.com/t/font_8d5l8fzk5b87iudi.js',
  });
  return (
    <div>
      <Icon type="loading" style={{ color: '#1890ff' }} />
      <Icon material="face" style={{ color: '#f66' }} rotate={90} />
      <Icon material="face" rotate={90} />
      <IconFont type="icon-twitter" spin />
      <IconFont type="icon-tuichu" rotate={90} />
    </div>
  );
};
