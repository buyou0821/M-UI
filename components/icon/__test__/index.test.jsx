import * as React from 'react';
import Icon from '..';
import { render, mount } from 'enzyme';

describe('Icon', () => {
  it('should render material icon', () => {
    const wrapper = render(<Icon material="face" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should support iconfont', () => {
    const IconFont = Icon.createFromIconfont({
      scriptUrl: '//at.alicdn.com/t/font_8d5l8fzk5b87iudi.js',
    });
    const wrapper = render(<IconFont type="icon-twitter" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should support onClick', () => {
    const fn = jest.fn();
    const wrapper = mount(<Icon material="face" onClick={fn} />);
    wrapper.simulate('click');
    expect(fn).toBeCalled();
  });
});
