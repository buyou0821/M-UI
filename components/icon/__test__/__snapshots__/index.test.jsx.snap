// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render material icon 1`] = `
<i
  class="muiicon material-icons"
>
  face
</i>
`;

exports[`Icon should support iconfont 1`] = `
<i
  class="muiicon"
>
  <svg
    fill="currentColor"
    height="1em"
    width="1em"
  >
    <use
      href="#icon-twitter"
    />
  </svg>
</i>
`;
